src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/ServiceRequestAPIHandlerTest.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.hmcts.reform.civil.handler.callback.camunda.payment;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.civil.callback.CallbackParams;
+import uk.gov.hmcts.reform.civil.handler.callback.BaseCallbackHandlerTest;
+import uk.gov.hmcts.reform.civil.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.hearing.HFPbaDetails;
+import uk.gov.hmcts.reform.civil.sampledata.CaseDataBuilder;
+import uk.gov.hmcts.reform.civil.service.PaymentsService;
+import uk.gov.hmcts.reform.civil.service.Time;
+import uk.gov.hmcts.reform.payments.response.PaymentServiceResponse;
+
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.CREATE_SERVICE_REQUEST_API;
+
+@SpringBootTest(classes = {
+    ServiceRequestAPIHandler.class,
+    JacksonAutoConfiguration.class,
+    CaseDetailsConverter.class
+})
+public class ServiceRequestAPIHandlerTest extends BaseCallbackHandlerTest {
```

Comment:
Please add a test for the `camundaActivityId` method as well. I know it seems trivial, but it can intercept typos or people modifying values they shouldn't be touching.

Also: the `catch` clause of `makePaymentServiceReq` is not intercepted by your tests, meaning you are only testing the happy path. Please add a test that simulates a `FeignException` to verify that everything is correct also in that case.

Reply:
A note: using IntelliJ's feature to run tests with coverage will help you detect these situations. After running the tests in that mode, IntelliJ will highlight in green, yellow or red the lines that have been hit during your tests. This will guide you to getting to 100% coverage.